### DISCUSSION TITLE
Using ENV's in .conf files

### DISCUSSION THREAD

**Discussion by NonaryR** [vesoft-inc/nebula#4225] [Q&A]
**General Question**

Hello! I can't find answer for this simple question - can I use environment variables for passing it into .conf files?
I am using nomad for orchestration, and ENV's is really a common way to configure all my services.
Thank you!

**Comment by jamieliu1023** [comment]
HI @NonaryR Thanks for bringing this up! I have converted this issue to a Discussion thread since it is actually a question. If any feature request come out of this question, I will raise an issue and relate it to this thread. That being said,  @wey-gu could you please help look at this?

**Comment by wey-gu** [comment]
Wow, exciting to have nomad hacker in nebula community, that's possible!

I used have a sketch here https://www.siwei.io/en/sketches/nebula-config-explained/nebula-config-explained.webp 

Nebula's config could be passed with command arguments and that is exactly how nebula docker compose is maintainering configurations with docker-compose file(for k8s, they were done via the conf file though).

https://github.com/vesoft-inc/nebula-docker-compose/blob/a0ae7ff6f840903497122b0424be4c68c2de1363/docker-compose.yaml#L8-12

Thus in theory, you could pass env variables to its command arguments in this format.
